.gitpod.yml

```diff
@@ -1,31 +1,19 @@
----
-# based on https://github.com/gitpod-io/template-docker-compose
+image: nfcore/gitpod:latest
 
-tasks:
-  - init: docker-compose pull
-  - command: chmod a+x /workspace/nf-co.re && docker-compose up
+vscode:
+  extensions: # based on nf-core.nf-core-extensionpack
+    - EditorConfig.EditorConfig              # override user/workspace settings with settings found in .editorconfig files
+    - Gruntfuggly.todo-tree                  # Display TODO and FIXME in a tree view in the activity bar
+    - mechatroner.rainbow-csv                # Highlight columns in csv files in different colors
+    # - nextflow.nextflow                      # Nextflow syntax highlighting
+    - oderwat.indent-rainbow                 # Highlight indentation level
 
 ports:
-  - port: 8888
-    onOpen: open-preview
-
-vscode:
-  extensions:
-    - ms-azuretools.vscode-docker
-    - esbenp.prettier-vscode
-github:
-  prebuilds:
-    # enable for the default branch (defaults to true)
-    master: true
-    # enable for all branches in this repo (defaults to false)
-    branches: false
-    # enable for pull requests coming from this repo (defaults to true)
-    pullRequests: true
-    # enable for pull requests coming from forks (defaults to false)
-    pullRequestsFromForks: true
-    # add a check to pull requests (defaults to true)
-    addCheck: true
-    # add a "Review in Gitpod" button as a comment to pull requests (defaults to false)
-    addComment: false
-    # add a "Review in Gitpod" button to the pull request's description (defaults to false)
-    addBadge: true
\ No newline at end of file
+- port: 8080
+  onOpen: ignore
+tasks:
+- init: git checkout gitpod_exercises
+- command: |
+        parsed=`echo $url | tr -s "[" "/" | tr -s "+" "."`
```

Comment:
So this parses the string 
```
https://gitpod.io/#url=nf-co+re[usage[tutorials[nf_core_usage_tutorial/https://github.com/nf-core/nf-co.re/blob/gitpod_config_training/
```
Parts of the URL:
```
https://gitpod.io/#                                              // Gitpod prefix
url=nf-co+re[usage[tutorials[nf_core_usage_tutorial              // Value pair, separate by commas
https://github.com/nf-core/nf-co.re/blob/gitpod_config_training/ // Repo + branch to open
```

Does URL encoding work (it says the values are URL encoded)? https://www.urlencoder.org/
```
https://carpentries-incubator.github.io/workflows-nextflow/01-getting-started-with-nextflow/index.html
```
becomes
```
https%3A%2F%2Fcarpentries-incubator.github.io%2Fworkflows-nextflow%2F01-getting-started-with-nextflow%2Findex.html
```
How can I test this? I tried editing the other page https://github.com/nf-core/nf-co.re/pull/1094/files
but then I can't get to the button in the preview browser. 

We may just be over complicating things here though.
What would be nice is just:
<img width="1437" alt="teaching_view" src="https://user-images.githubusercontent.com/15652539/159077744-53420d12-5c8c-4ede-97e8-68e216de1987.png">

So a preview panel with the website (using `gp preview URL` and it's navigable)
And then as large a terminal as possible. I moved the terminal position to the right to have them side by side.
( Menu > View > Appearance > Panel position > right ).
Then they can do all their work in an exercises folder and not have to worry about all the other distractions along the side.

Reply:
I was too focused on carpentries material before and forgot we have two use cases here. 

For the nf-core training material we want buttons on each page as they are standalone training guides, so we should definitely see if URL encoding is sufficient to open the correct URL with gp preview. 

The Carpentries Workshop though just needs to start from the setup page and can have a simpler link.